rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -743,10 +733,10 @@ public void finishNodeSourceShutdown(Client initiator) {
      * @return the list of deploying nodes handled by the infrastructure manager
      */
     @ImmediateService
-    public LinkedList<RMDeployingNode> getDeployingNodes() {
-        LinkedList<RMDeployingNode> result = new LinkedList<>();
-        result.addAll(this.infrastructureManager.getDeployingNodes());
-        return result;
+    public List<RMNodeData> getDeployingNodes() {
```

Comment:
You are right, I agree it is not the right type to expose. I just need to check whether this could be executed before RMDeployingNode are rebuilt from RMNodeData.